### PR TITLE
chore(ts): refactor and migrate StackTraceMessage to TypeScript

### PR DESCRIPTION
**Pull request by etr2460**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
In advance of further error message work, migrating the StackTraceMessage to TypeScript and rename it to a more clear (in my opinion) ErrorMessageWithStackTrace.

Also adds a new `npm run type` command and adds it to `npm run lint` to ensure that typescript errors are caught by the linter

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Both screenshots after the refactor:
Screen Shot 2020-04-27 at 11 06 43 AM
Screen Shot 2020-04-27 at 11 08 32 AM

### TEST PLAN
Ensure error messages with and without stacktraces still render properly
run `npm run lint`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
to: @graceguo-supercat @ktmud @rusackas @nytai

### CODE REVIEW

File: superset-frontend/src/components/ErrorMessageWithStackTrace.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
```
Comment (etr2460) [inline superset-frontend/src/components/ErrorMessageWithStackTrace.tsx:1]: this file is moved from the old jsx file, major changes are making it a functional component and replacing the state object with a useState hook. Note that the old prop `showStackTrace` was unused except to initialize the state, so i removed it

File: superset-frontend/src/components/ErrorMessageWithStackTrace.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// @ts-ignore
+import { Alert, Collapse } from 'react-bootstrap';
+
+interface Props {
+  message: string;
+  link?: string;
+  stackTrace?: string;
+}
+
+function ErrorMessageWithStackTrace({ message, link, stackTrace }: Props) {
```
Comment (ktmud) [inline superset-frontend/src/components/ErrorMessageWithStackTrace.tsx:29]: Can `expore default` here.

**Review by ktmud — APPROVED** [review]
lgtm!

File: superset-frontend/package.json [hunk 3]
```diff
@@ -10,16 +10,17 @@
...
     "dev-server": "NODE_ENV=development BABEL_ENV=development node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --mode=development --progress",
     "prod": "node --max_old_space_size=4096 ./node_modules/webpack/bin/webpack.js --mode=production --colors --progress",
     "build-dev": "cross-env NODE_OPTIONS=--max_old_space_size=8192 NODE_ENV=development webpack --mode=development --colors --progress",
     "build-instrumented": "cross-env NODE_ENV=development BABEL_ENV=instrumented webpack --mode=development --colors --progress",
     "build": "cross-env NODE_OPTIONS=--max_old_space_size=8192 NODE_ENV=production webpack --mode=production --colors --progress",
-    "lint": "eslint --ignore-path=.eslintignore --ext .js,.jsx,.ts,.tsx .",
+    "lint": "eslint --ignore-path=.eslintignore --ext .js,.jsx,.ts,.tsx . && npm run type",
     "prettier-check": "prettier --check '{src,stylesheets}/**/*.{css,less,sass,scss}'",
-    "lint-fix": "eslint --fix --ignore-path=.eslintignore --ext .js,.jsx,.ts,tsx . && npm run clean-css",
+    "lint-fix": "eslint --fix --ignore-path=.eslintignore --ext .js,.jsx,.ts,tsx . && npm run clean-css && npm run type",
```
Comment (rusackas) [inline superset-frontend/package.json:23]: For a brief moment, I dreamed of a world where you could `--fix` your way into TypeScript :)

Thanks for adding this checking.

File: superset-frontend/src/components/ErrorMessageWithStackTrace.tsx (hunk 2 above)
Comment (etr2460) [reply superset-frontend/src/components/ErrorMessageWithStackTrace.tsx]: fair enough, done

**Review by rusackas — APPROVED** [review]
LGTM

**Pull request merged**